### PR TITLE
gitflow-v1

### PR DESCRIPTION
**Pull request by aslanmehmetsalih**
- add gitflow images
- update README.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,105 @@
...
+
+
+<a name="hotfix"></a>
+## Hotfix Branches
+
+![](images/hotfix.svg)
+
+
+* The `hotfix` branches are used for quick bugfix on production releases.
+* Each new `hotfix` should be created from `release`.
```
Comment (alihanaktay) [inline README.md]: I think hotfix branches should be created from master branch

File: README.md [hunk 2]
```diff
@@ -1 +1,105 @@
...
+* Branch name should be start with `hotfix/` and end with task name.
```
Comment (alihanaktay) [inline README.md]: If branch end with version number like release branches we can trigger the CI system with this version number. For my opinion there is no need for task name instead we can use version number

**Review by alihanaktay — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (aslanmehmetsalih) [reply README.md]: fix typo

File: README.md (hunk 1 above)
Comment (aslanmehmetsalih) [reply README.md]: fix typo

File: README.md [hunk 3]
```diff
@@ -1 +1,105 @@
...
+1. [References](#references)
+
+
+<a name="master-develop"></a>
+## Master & Develop Branches
+
+![](images/master-develop.svg)
+
+We have two main branches. These are `master` and `develop`.
+* The `master` branch store the latest release codebase.
```
Comment (utkuyildirim) [inline README.md]:
```suggestion
* The `master` branch store the latest released codebase.
```

File: README.md [hunk 4]
```diff
@@ -1 +1,105 @@
...
+* The `develop` branch store the updated codebase.
```
Comment (utkuyildirim) [inline README.md]:
```suggestion
* The `develop` branch store the latest codebase.
```

File: README.md [hunk 5]
```diff
@@ -1 +1,105 @@
...
+
+
+<a name="feature-epic"></a>
+## Feature & Epic Branches
+
+![](images/feature-epic.svg)
+
+### Small Tasks
+* Each new `small` `feature` should be created from `develop`.
+* Branch name should be start with `feature/` and end with task name.
```
Comment (utkuyildirim) [inline README.md]: We should to use feature/ISSUECODE at a minimum.

I strongly suggest using pattern below to identify the issue.
feature/ISSUECODE-short-task-name

File: README.md [hunk 6]
```diff
@@ -1 +1,105 @@
...
+
+
+
+<a name="release-bugfix"></a>
+## Release & Bugfix Branches
+
+![](images/release-bugfix.svg)
+
+> * ***Only `bugfix` can be made on the `release` branch.***
+> * ***No new feature can be added to after branch created.***
```
Comment (utkuyildirim) [inline README.md:54]:
```suggestion
> * ***No new feature can be added to after branch created.***
> * ***If a new feature needs to be added, new release branch with increased version must be created.***
```

**Review by utkuyildirim — CHANGES_REQUESTED** [review]
There is a general question should we include ISSUE IDs from the Project Management system to feature, epic and bugfix branches?

**Review by utkuyildirim — APPROVED** [review]

**Pull request merged**